diff --git a/browser/app/nsBrowserApp.cpp b/browser/app/nsBrowserApp.cpp
index 56fb4bb..1975b2a 100644
--- a/browser/app/nsBrowserApp.cpp
+++ b/browser/app/nsBrowserApp.cpp
@@ -187,6 +187,18 @@ static int do_main(int argc, char* argv[], nsIFile *xreDirectory)
   nsresult rv;
   uint32_t mainFlags = 0;

+  bool argBrowser = false;
+  bool argChrome = false;
+
+  for (int z=0; z<argc; z++) {
+    if(IsArg(argv[z], "browser")) {
+      argBrowser = true;
+    }
+    else if(IsArg(argv[z], "chrome")) {
+      argChrome = true;
+    }
+  }
+
   // Allow firefox.exe to launch XULRunner apps via -app <application.ini>
   // Note that -app must be the *first* argument.
   const char *appDataFile = getenv("XUL_APP_FILE");
@@ -219,7 +231,31 @@ static int do_main(int argc, char* argv[], nsIFile *xreDirectory)
     argv += 2;
     argc -= 2;
   }
+  else if (!(argBrowser || argChrome)){
+   //start in 'xulrunner' replacement mode
+   char overrideIniPath[MAXPATHLEN];
+   nsCOMPtr<nsIFile> browserSubdir;
+    xreDirectory->Clone(getter_AddRefs(browserSubdir));
+    browserSubdir->Append(NS_LITERAL_STRING("browser"));
+    browserSubdir->Append(NS_LITERAL_STRING("application.ini"));
+    //create C-str
+   nsCString path;
+   browserSubdir->GetNativePath(path);
+
+    sprintf(overrideIniPath, "%s", ToNewCString(path));
+   rv = XRE_GetFileFromPath(overrideIniPath, getter_AddRefs(appini));
+    if (NS_FAILED(rv)) {
+      Output("application.ini path not recognized: '%s'", overrideIniPath);
+      return 255;
+    }

+    char appEnv[MAXPATHLEN];
+    snprintf(appEnv, MAXPATHLEN, "XUL_APP_FILE=%s", overrideIniPath);
+    if (putenv(appEnv)) {
+      Output("Couldn't set %s.\n", appEnv);
+      return 255;
+    }
+  }
   if (appini) {
     nsXREAppData *appData;
     rv = XRE_CreateAppData(appini, &appData);

diff --git a/browser/confvars.sh b/browser/confvars.sh
index d9e5d54..f15d2af 100755
--- a/browser/confvars.sh
+++ b/browser/confvars.sh
@@ -23,7 +23,7 @@ if test "$OS_ARCH" = "WINNT"; then
   fi
 fi

-MOZ_CHROME_FILE_FORMAT=omni
+MOZ_CHROME_FILE_FORMAT=flat
 MOZ_DISABLE_EXPORT_JS=1
 MOZ_SAFE_BROWSING=1
 MOZ_SERVICES_COMMON=1
