diff --git a/browser/app/nsBrowserApp.cpp b/browser/app/nsBrowserApp.cpp
index ba0eb60..a789826 100644
--- a/browser/app/nsBrowserApp.cpp
+++ b/browser/app/nsBrowserApp.cpp
@@ -212,7 +212,31 @@ static int do_main(int argc, char* argv[], nsIFile *xreDirectory)
     argv += 2;
     argc -= 2;
   }
+  else if (!(argc > 1 && IsArg(argv[1], "browser"))){
+  	//hack
+  	char overrideIniPath[MAXPATHLEN];
+  	nsCOMPtr<nsIFile> browserSubdir;
+    xreDirectory->Clone(getter_AddRefs(browserSubdir));
+    browserSubdir->Append(NS_LITERAL_STRING("browser"));
+    browserSubdir->Append(NS_LITERAL_STRING("application.ini"));
+    //create C-str
+  	nsCString path;
+  	browserSubdir->GetNativePath(path);
+
+	sprintf(overrideIniPath, "%s", ToNewCString(path));
+  	rv = XRE_GetFileFromPath(overrideIniPath, getter_AddRefs(appini));
+    if (NS_FAILED(rv)) {
+      Output("application.ini path not recognized: '%s'", overrideIniPath);
+      return 255;
+    }
 
+    char appEnv[MAXPATHLEN];
+    snprintf(appEnv, MAXPATHLEN, "XUL_APP_FILE=%s", overrideIniPath);
+    if (putenv(appEnv)) {
+      Output("Couldn't set %s.\n", appEnv);
+      return 255;
+    }
+  }
   if (appini) {
     nsXREAppData *appData;
     rv = XRE_CreateAppData(appini, &appData);
diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
index 10bc6a9..f55cc1e 100644
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -32,10 +32,10 @@ pref("extensions.strictCompatibility", false);
 
 // Specifies a minimum maxVersion an addon needs to say it's compatible with
 // for it to be compatible by default.
-pref("extensions.minCompatibleAppVersion", "4.0");
+//pref("extensions.minCompatibleAppVersion", "4.0");
 // Temporary preference to forcibly make themes more safe with Australis even if
 // extensions.checkCompatibility=false has been set.
-pref("extensions.checkCompatibility.temporaryThemeOverride_minAppVersion", "29.0a1");
+//pref("extensions.checkCompatibility.temporaryThemeOverride_minAppVersion", "29.0a1");
 
 // Preferences for AMO integration
 pref("extensions.getAddons.cache.enabled", true);
diff --git a/browser/confvars.sh b/browser/confvars.sh
index d9e5d54..f15d2af 100755
--- a/browser/confvars.sh
+++ b/browser/confvars.sh
@@ -23,7 +23,7 @@ if test "$OS_ARCH" = "WINNT"; then
   fi
 fi
 
-MOZ_CHROME_FILE_FORMAT=omni
+MOZ_CHROME_FILE_FORMAT=flat
 MOZ_DISABLE_EXPORT_JS=1
 MOZ_SAFE_BROWSING=1
 MOZ_SERVICES_COMMON=1
