diff --git a/browser/app/nsBrowserApp.cpp b/browser/app/nsBrowserApp.cpp
index 56fb4bb..1975b2a 100644
--- a/browser/app/nsBrowserApp.cpp
+++ b/browser/app/nsBrowserApp.cpp
@@ -187,6 +187,18 @@ static int do_main(int argc, char* argv[], nsIFile *xreDirectory)
   nsresult rv;
   uint32_t mainFlags = 0;

+  bool argBrowser = false;
+  bool argChrome = false;
+
+  for (int z=0; z<argc; z++) {
+    if(IsArg(argv[z], "browser")) {
+      argBrowser = true;
+    }
+    else if(IsArg(argv[z], "chrome")) {
+      argChrome = true;
+    }
+  }
+
   // Allow firefox.exe to launch XULRunner apps via -app <application.ini>
   // Note that -app must be the *first* argument.
   const char *appDataFile = getenv("XUL_APP_FILE");
@@ -219,7 +231,31 @@ static int do_main(int argc, char* argv[], nsIFile *xreDirectory)
     argv += 2;
     argc -= 2;
   }
+  else if (!(argBrowser || argChrome)){
+   //start in 'xulrunner' replacement mode
+   char overrideIniPath[MAXPATHLEN];
+   nsCOMPtr<nsIFile> browserSubdir;
+    xreDirectory->Clone(getter_AddRefs(browserSubdir));
+    browserSubdir->Append(NS_LITERAL_STRING("browser"));
+    browserSubdir->Append(NS_LITERAL_STRING("application.ini"));
+    //create C-str
+   nsCString path;
+   browserSubdir->GetNativePath(path);
+
+    sprintf(overrideIniPath, "%s", ToNewCString(path));
+   rv = XRE_GetFileFromPath(overrideIniPath, getter_AddRefs(appini));
+    if (NS_FAILED(rv)) {
+      Output("application.ini path not recognized: '%s'", overrideIniPath);
+      return 255;
+    }

+    char appEnv[MAXPATHLEN];
+    snprintf(appEnv, MAXPATHLEN, "XUL_APP_FILE=%s", overrideIniPath);
+    if (putenv(appEnv)) {
+      Output("Couldn't set %s.\n", appEnv);
+      return 255;
+    }
+  }
   if (appini) {
     nsXREAppData *appData;
     rv = XRE_CreateAppData(appini, &appData);

diff --git a/browser/confvars.sh b/browser/confvars.sh
index d9e5d54..f15d2af 100755
--- a/browser/confvars.sh
+++ b/browser/confvars.sh
@@ -23,7 +23,7 @@ if test "$OS_ARCH" = "WINNT"; then
   fi
 fi

-MOZ_CHROME_FILE_FORMAT=omni
+MOZ_CHROME_FILE_FORMAT=flat
 MOZ_DISABLE_EXPORT_JS=1
 MOZ_SAFE_BROWSING=1
 MOZ_SERVICES_COMMON=1
diff --git a/browser/app/macbuild/Contents/Info.plist.in b/browser/app/macbuild/Contents/Info.plist.in
index ed2cf8d..19fce7d 100644
--- a/browser/app/macbuild/Contents/Info.plist.in
+++ b/browser/app/macbuild/Contents/Info.plist.in
@@ -143,11 +143,11 @@
    </dict>
  </array>
  <key>CFBundleExecutable</key>
- <string>firefox</string>
+ <string>SiteFusion</string>
  <key>CFBundleGetInfoString</key>
  <string>%MAC_APP_NAME% %APP_VERSION%</string>
  <key>CFBundleIconFile</key>
- <string>firefox.icns</string>
+ <string>SiteFusion.icns</string>
  <key>CFBundleIdentifier</key>
  <string>%MOZ_MACBUNDLE_ID%</string>
  <key>CFBundleInfoDictionaryVersion</key>
diff --git a/browser/app/Makefile.in b/browser/app/Makefile.in
index d1c6eba..bda8338 100644
--- a/browser/app/Makefile.in
+++ b/browser/app/Makefile.in
@@ -8,7 +8,7 @@ dist_dest   = $(DIST)/$(MOZ_MACBUNDLE_NAME)
 AB_CD = en-US

 DEFINES += \
-  -DFIREFOX_ICO='"$(DIST)/branding/firefox.ico"' \
+  -DFIREFOX_ICO='"$(DIST)/branding/sitefusion.ico"' \
   -DDOCUMENT_ICO='"$(DIST)/branding/document.ico"' \
   -DNEWWINDOW_ICO='"$(DIST)/branding/newwindow.ico"' \
   -DNEWTAB_ICO='"$(DIST)/branding/newtab.ico"' \
